cpp/cuio/kafka/include/kafka_consumer.hpp

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <librdkafka/rdkafkacpp.h>
+#include <sys/time.h>
+#include <algorithm>
+#include <cudf/io/datasource.hpp>
+#include <map>
+#include <memory>
+#include <string>
+
+namespace cudf {
+namespace io {
+namespace external {
+namespace kafka {
+
+/**
+ * @brief libcudf external datasource for Apache Kafka
+ **/
+class kafka_consumer : public cudf::io::datasource {
+  /**
+   * @brief Implementation for holding kafka messages
+   **/
+  class message_buffer : public buffer {
+   public:
+    message_buffer(uint8_t *data, size_t size) : _data(data), _size(size) {}
```

Comment:
I'm all for that. Exactly which header are you thinking?

Reply:
Ah, missed this one.
I think it would be best to add this class definition to the end of `cpp/include/cudf/io/datasource.hpp`
I think the code is fine as-is, just rename the class and update the doxy comments.

I would appreciate it if you could also update `TestSource` in `cpp/tests/io/csv_test.cu` to use this new class :)